cpp/src/arrow/testing/gtest_util.cc

```diff
@@ -382,11 +382,7 @@ void SleepFor(double seconds) {
 // Extension types
 
 bool UUIDType::ExtensionEquals(const ExtensionType& other) const {
-  const auto& other_ext = static_cast<const ExtensionType&>(other);
-  if (other_ext.extension_name() != this->extension_name()) {
-    return false;
-  }
-  return true;
+  return (other.extension_name() == this->extension_name());
 }
```

Comment:
I wonder if this should be the default implementation of `ExtensionEquals`

Reply:
Created a follow-up JIRA https://issues.apache.org/jira/browse/ARROW-8143